src/main/java/kitchenpos/menu/domain/Menu.java

```diff
@@ -1,28 +1,30 @@
-package kitchenpos.domain;
+package kitchenpos.menu.domain;
 
+import javax.persistence.*;
 import java.math.BigDecimal;
 import java.util.List;
 import java.util.Objects;
 
+@Entity
 public class Menu {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String name;
-    private BigDecimal price;
+    private Price price;
     private Long menuGroupId;
-    private List<MenuProduct> menuProducts;
 
-    public Menu(String name, BigDecimal price, Long menuGroupId, List<MenuProduct> menuProducts) {
-        this.name = name;
-        this.price = price;
-        this.menuGroupId = menuGroupId;
-        this.menuProducts = menuProducts;
+    @Embedded
+    private MenuProducts menuProducts = new MenuProducts();
+
+    protected Menu() {
     }
 
-    public Menu(Long id, String name, BigDecimal price, Long menuGroupId) {
-        this.id = id;
+    public Menu(String name, BigDecimal price, Long menuGroupId, List<MenuProduct> menuProducts) {
```

Comment:
```suggestion
    public Menu(String name, Price price, Long menuGroupId, List<MenuProduct> menuProducts) {
```
와는 어떤 차이가 있을까요? 어떤 방향이더 좋은 방향일지도 고민해보면 좋을 것 같아요.

Reply:
음.. 저는 생성을 좀 더 쉽게 하기 위해서 BigDecimal을 사용했는데 Price를 넘겨주게 된다면 Price에 대한 예외처리(가격이 0원 이상이다) 같은 것들을 Menu에서는 신경쓰지 않고 사용할 수 있을 것 같고 Price가 BigDecimal이 아닌 다른 객체로 변경되더라도 Menu에서는 실제 내부 구현체를 모르기 때문에 코드 변경도 적을 것 같습니다.